### PR TITLE
Index once

### LINKED ISSUE #6134: [unique: true] creates two unique indexes on a column
For some a reason column option `unique: true` creates two unique indexes on a column. How to prevent it? Please take a look at an `email` field:

```
const mysql = require('../getConnection').mysql; // configured and connected sequelize here

const modelAttrs = {
    fname: {
        type: mysql.Sequelize.STRING(255),
        field: 'fname',
        allowNull: true,
        defaultValue: null,
    },
    email: {
        type: mysql.Sequelize.STRING(254),
        field: 'email',
        allowNull: true,
        defaultValue: null,
        unique: true,      // <<== here an email column is unique
        validate: {
            isEmail: true,
        }
    },
    time_zone: {
        type: mysql.Sequelize.STRING(100),
        field: 'time_zone',
        allowNull: true,
        defaultValue: null,
    }
};

const modelOpts = {
    freezeTableName: true,
    tableName: 'UserProfile',
    timestamps: true,
}

const UserProfile = mysql.define('UserProfile', modelAttrs, modelOpts);

mysql.sync({ force: true });
```

And here is what is actually created, one unique key on `email` is redundant:

```
CREATE TABLE `userprofile` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `fname` varchar(255) DEFAULT NULL,
  `lname` varchar(255) DEFAULT NULL,
  `full_name` varchar(500) DEFAULT NULL,
  `email` varchar(254) DEFAULT NULL,
  `accept_ts` tinyint(1) NOT NULL DEFAULT '0',
  `time_zone` varchar(100) DEFAULT NULL,
  `createdAt` datetime NOT NULL,
  `updatedAt` datetime NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `email` (`email`), //      <<==  One
  UNIQUE KEY `UserProfile_email_unique` (`email`)  //     <<==  Two ??
) ENGINE=InnoDB DEFAULT CHARSET=utf8
```

Why does it happen? How to prevent it?

### PR DESCRIPTION
**Pull request by BridgeAR**
### Pull Request check-list
- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does your issue contain a link to existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Have you added an entry under `Future` in the changelog?
### Description of change

Create only one unique index instead of two

Fixes #6134

### CODE REVIEW

**Comment by BridgeAR** [comment]
I can't properly debug mssql

File: test/unit/model/sync.test.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+var chai = require('chai')
+  , expect = chai.expect
+  , sinon = require('sinon')
+  , Support   = require(__dirname + '/../support')
+  , DataTypes = require('../../../lib/data-types')
+  , current   = Support.sequelize
+  , Promise = current.Promise;
+
+describe(Support.getTestDialectTeaser('Model'), function() {
+  describe.only('sync', function () {
```
Comment (mickhansen) [inline test/unit/model/sync.test.js]: !

Comment (BridgeAR) [inline test/unit/model/sync.test.js]: Oops. Fixed

**Comment by janmeier** [comment]
It think we can just skip this test for mssql https://github.com/sequelize/sequelize/blob/6177a95ddcc598fe60a0cbc12273fb33371de22c/test/integration/instance.validations.test.js#L82

You should be able to debug mssql just by running `npm run test-integration-mssql` (outside of docker) - that will hit out public mssql servers.

I don't think we should try to parse which field from this: 
```js
Violation of UNIQUE KEY constraint 'UQ__models__38751322B57C6E6B'. Cannot insert duplicate key in object 'dbo.models'. The duplicate key value is (unique name one).
```

Theoretically I guess we can infer the column from the value, but noooo....

**Comment by BridgeAR** [comment]
I gave up looking into this. I do not know enough about mssql.

Something does not work with mssql and findOrCreate anymore after using only the normal unique index. The only real change I see is that the returned err.fields look like: `UQ__Users__387513223698177A` instead of `uniqueName`. It seems like there is a deadlock while running the find after the unique constraint error.

**Comment by mccabemj** [comment]
Went to fix this issue #6134 and saw this PR. My mssql tests pass with that change to /lib/model.js. What was the issue?

**Comment by janmeier** [comment]
@mccabemj I don't remember - You are welcome to open a new PR, perhaps it was a flaky test that has since been fixed / remove

**Pull request closed without merging**